core/drand.go

```diff
@@ -223,12 +223,20 @@ func (d *Drand) WaitDKG() (*key.Group, error) {
 
 // StartBeacon initializes the beacon if needed and launch a go
 // routine that runs the generation loop.
-func (d *Drand) StartBeacon(catchup bool) {
-	b, err := d.newBeacon()
+func (d *Drand) StartBeacon(catchup bool, oldGroup string, skipValidation bool) {
+	b, err := d.newBeacon(skipValidation)
 	if err != nil {
 		d.log.Error("init_beacon", err)
 		return
 	}
+	if oldGroup != "" {
+		var grp = new(key.Group)
+		if err := key.Load(oldGroup, grp); err != nil {
+			d.log.Error("beacon_start", time.Now(), "couldn't load old group", err)
```

Comment:
Missing return?

Reply:
i think i'm okay with an error and continued execution. 🤷